### PR TITLE
feat(pds-ember): add Pds::Modal

### PR DESCRIPTION
**Pull request by CITguy**
JIRA: PDS-138

### Variants
_Plain (default)_
Modal - Basic - Plain

_Error_
Modal - Basic - Error

_Info_
Modal - Basic - Info

_Success_
Modal - Basic - Success

_Warning_
Modal - Basic - Warning

### Configuration and Usage
_Custom Icon_
Modal - Basic - Plain - Custom Icon

_Empty_
Modal - Basic - Plain - Empty

_Lengthy Content_
Modal - Basic - Plain - Lengthy Content

_Lengthy Title_
Modal - Basic - Plain - Lengthy Title

### CODE REVIEW

File: packages/pds-ember/addon/components/pds/modal/index.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  }
+
+  @action
+  close() {
+    this.onClose()
+  }
+
+  @action
+  _handleKeyup(evt) {
+    if (evt.keyCode === Esc) {
```
Comment (CITguy) [inline packages/pds-ember/addon/components/pds/modal/index.js:59]: Apparently `evt.keyCode` is deprecated in favor of `evt.key`.

```javascript
if (evt.key === 'Escape') {
  this.close()
}
```

File: packages/pds-ember/addon/utils/keyCodes.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
+export const Alt = 18
```
Comment (CITguy) [inline packages/pds-ember/addon/utils/keyCodes.js:1]: Remove file, it's not needed for `evt.key` evaluation.

File: packages/pds-ember/app/styles/pds/components/modal/__config.scss [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+
+@mixin _layout--dialog {
+  &__dialog {
+    display: flex;
+    flex-direction: column;
+    max-height: 100%;
+    width: 30rem; // ~480px // TODO: verify with Design
+
+    > * {
+      padding: 0.75rem 1rem;
```
Comment (CITguy) [inline packages/pds-ember/app/styles/pds/components/modal/__config.scss:80]: use theme or scale variables

File: packages/pds-ember/app/styles/pds/components/modal/__config.scss [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+    flex-shrink: 0;
+    order: 3;
+  }
+}
+
+@mixin _layout--dialog-header {
+  &__header {
+    align-items: center;
+    display: grid;
+    grid-gap: 0 0.25rem;
```
Comment (CITguy) [inline packages/pds-ember/app/styles/pds/components/modal/__config.scss:106]: use theme or scale variable

File: packages/pds-ember/app/styles/pds/components/modal/__config.scss [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+@mixin layout {
+  /* [debug] #{$_module}@layout */
+  @include _layout--root;
+  @include _layout--dialog;
+  @include _layout--dialog-header;
+}
+
+@mixin appearance {
+  /* [debug] #{$_module}@appearance */
+  background-color: rgba(color.$ui-gray-900, 0.86); // "blanket"
```
Comment (CITguy) [inline packages/pds-ember/app/styles/pds/components/modal/__config.scss:142]: convert value to variable

**Review by CITguy — COMMENTED** [review]
consider replacing `<M.Title>` with `@title`

File: packages/pds-ember/addon/components/pds/modal/index.js [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+  }
+
+  @action
+  close() {
+    this.onClose()
+  }
+
+  @action
+  _handleKeyup(evt) {
+    if (evt.key === 'Escape') {
```
Comment (CITguy) [inline packages/pds-ember/addon/components/pds/modal/index.js:76]: See https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/key/Key_Values

File: packages/pds-ember/addon/components/pds/modal/index.js [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+
+  /**
+   * If left unset, icon will be applied based on the configured variant.
+   *
+   * @argument icon
+   * @type {string}
+   */
+  get iconType() {
+    let { icon } = this.args
+    let result = icon ? icon : ICON_TYPES[this.variant]
```
Comment (cbfx) [inline packages/pds-ember/addon/components/pds/modal/index.js:43]: amazing. this is perfect for the snapshot modals i'm working on now.

File: packages/pds-ember/app/styles/pds/components/modal/__config.scss [hunk 8]
```diff
@@ -0,0 +1,196 @@
...
+
+@mixin theme($styles) {
+  /* [debug] #{$_module}@theme */
+  @include Component.theme(modal, $styles);
+}
+
+$PLAIN: (
+  header-backgroundColor: color.$ui-gray-050,
+  header-borderColor: theme.$borderColor,
+  title-color: color.$ui-gray-900,
```
Comment (cbfx) [inline packages/pds-ember/app/styles/pds/components/modal/__config.scss:35]: `icon-color` should come before `title-color` here.

Comment (CITguy) [reply packages/pds-ember/app/styles/pds/components/modal/__config.scss]: corrected

File: packages/pds-ember/addon/components/pds/modal/index.hbs [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+          @compact={{true}}
+          @iconStart="cancel-plain"
+          aria-label={{t 'pds.components.modal.close'}}
+          {{on 'click' this.close}}
+          data-test-modal-close-button
+        />
+      </div>
+    </header>
+
+    {{ yield (hash
```
Comment (cbfx) [inline packages/pds-ember/addon/components/pds/modal/index.hbs:40]: is this space intentional?

Comment (CITguy) [reply packages/pds-ember/addon/components/pds/modal/index.hbs:40]: auto generated by blueprints, has no affect on output

**Review by cbfx — APPROVED** [review]

**Pull request merged**